src/main/java/io/deephaven/csv/util/Util.java

```diff
@@ -0,0 +1,14 @@
+package io.deephaven.csv.util;
+
+public class Util {
+    /** Call Object.wait() but suppress the need to deal with checked InterruptedExceptions. */
+    public static void catchyWait(Object o) {
+        try {
+            o.wait();
+        } catch (InterruptedException ie) {
+            throw new RuntimeException(
+                    "Thread interrupted: probably cancelled in the CsvReader class due to some other exception.");
+        }
```

Comment:
IMO, we should be trying to push InterruptedException's as far up the stack as possible. It lets callers know "this call may block", and gives higher level logic (which may have more context) the ability to more gracefully deal with it.

I'm not sure it's worth piping through today, but I think we may need to revisit this in the future.